File: C++/rotateArr1.cpp
//Rotating using the same array or by using the 'temp' variable
#include<iostream>
using namespace std;

void printArr(int arr[], int size) {
    for(int i=0; i<size; i++) {
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

void RightRotateArr(int arr[], int size) {
    int temp = arr[size-1];
    for(int i=size-1; i>0; i--) {
        arr[i] = arr[i-1];
    }
    arr[0] = temp;
    cout<<"Right Rotated Array: "<<endl;
    printArr(arr, size);
}

void LeftRotateArr(int arr[], int size) {
    int temp = arr[0];
    for(int i=0; i<size-1; i++) {
        arr[i] = arr[i+1];
    }
    arr[size-1] = temp;
    cout<<"Left Rotated Array: "<<endl;
    printArr(arr, size);
}

int main() {
    int arr[] = {1,2,3,4,5};
    int size = sizeof(arr) / sizeof(arr[0]);
    cout<<"Original Array: "<<endl;
    printArr(arr, size);
    RightRotateArr(arr, size);
    LeftRotateArr(arr, size); //This will give back the original array
    return 0;
}

File: Python/rotateArr1.py
#//Rotating using the same array or by using the 'temp' variable
def RightRotateArr(arr, size):
    temp = arr[size-1]
    for i in range(size-1, 0, -1):
        arr[i] = arr[i-1]
    arr[0] = temp

def LeftRotateArr(arr, size):
    temp = arr[0]
    for i in range(0, size-1):
        arr[i] = arr[i+1]
    arr[size-1] = temp

def printArr(arr, size):
    for i in range(size):
        print(arr[i], end=" ")
    print("\n")

arr = [1,2,3,4,5]
size = len(arr)
print("Original Array: ")
printArr(arr, size)

RightRotateArr(arr, size)

print("Right Rotated Array: ")
printArr(arr, size)

LeftRotateArr(arr, size)

print("Left Rotated Array: ")
printArr(arr, size)